### PR TITLE
 i18n(it): translate docs/guides/backend/google-firebase.mdx

### PR DESCRIPTION
**Pull request by vclemenzi**
#### What kind of changes does this PR include?

- Translated content

#### Description

translated `docs/guides/backend/google-firebase.mdx`

### CODE REVIEW

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 1]
```diff
@@ -0,0 +1,915 @@
...
+</FileTree>
+
+## Aggiunta dell'autenticazione con Firebase
+
+### Prerequisiti
+
+- Un progetto Astro [inizializzato con Firebase](#inizializzazione-di-firebase-in-astro).
+- Un progetto Firebase con l'autenticazione tramite email/password abilitata nella console di Firebase in *Autenticazione > Metodo di accesso*.
+
+### Creazione dei endpoint del server per l'autenticazione
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:175]:
```suggestion
### Creazione degli endpoint del server per l'autenticazione
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 2]
```diff
@@ -0,0 +1,915 @@
...
+export const get: APIRoute = async ({ redirect, cookies }) => {
+  cookies.delete("session", {
+    path: "/",
+  });
+  return redirect("/signin");
+};
+```
+
+:::note
+Questa è un'implementazione di base del endpoint per la disconnessione. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:246]:
```suggestion
Questa è un'implementazione di base dell'endpoint per la disconnessione. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 3]
```diff
@@ -0,0 +1,915 @@
...
+      "Qualcosa è andato storto",
+      { status: 400 }
+    );
+  }
+  return redirect("/signin");
+};
+```
+
+:::note
+Questa è un'implementazione di base del endpoint per la registrazione. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:290]: I found multiple occurrences of `del endopoint`... Shouldn't it be `dell'endpoint` instead? Same for `al endpoint` -> `all'endpoint`?
E.g. at line 554 there's an example of what I think it's the correct form.

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 4]
```diff
@@ -0,0 +1,915 @@
...
+<Layout title={friend.name}>
+  <h1>{friend.name}</h1>
+  <p>Età: {friend.age}</p>
+  <p>Migliore amico: {friend.isBestFriend ? "Sì" : "No"}</p>
+</Layout>
+```
+
+#### Mostra una lista di record con un bottone per modificarli
+
+Infine, `src/pages/dashboard.astro` mostrerà una lista degli amici. Ogni amic avrà un link alla loro pagina e un bottone che li manderà sulla pagina per modificare l'amico.
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:846]:
```suggestion
Infine, `src/pages/dashboard.astro` mostrerà una lista degli amici. Ogni amico avrà un link alla loro pagina e un bottone che li manderà sulla pagina per modificare l'amico.
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 5]
```diff
@@ -0,0 +1,915 @@
...
+const friendsRef = db.collection("friends");
+const friendsSnapshot = await friendsRef.get();
+const friends = friendsSnapshot.docs.map((doc) => ({
+  id: doc.id,
+  ...doc.data(),
+})) as Friend[];
+---
+
+<Layout title="My friends">
+  <h1>Friends</h1>
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:871]:
```suggestion
  <h1>Amici</h1>
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 6]
```diff
@@ -0,0 +1,915 @@
...
+const db = getFirestore(app);
+const friendsRef = db.collection("friends");
+const friendsSnapshot = await friendsRef.get();
+const friends = friendsSnapshot.docs.map((doc) => ({
+  id: doc.id,
+  ...doc.data(),
+})) as Friend[];
+---
+
+<Layout title="My friends">
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:870]:
```suggestion
<Layout title="I miei amici">
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 7]
```diff
@@ -0,0 +1,915 @@
...
+  <h1>Friends</h1>
+  <ul>
+    {
+      friends.map((friend) => (
+        <li>
+          <a href={`/friend/${friend.id}`}>{friend.name}</a>
+          <span>({friend.age})</span>
+          <strong>{friend.isBestFriend ? "Bestie" : "Friend"}</strong>
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:878]:
```suggestion
          <strong>{friend.isBestFriend ? "Miglior amico" : "Amico"}</strong>
```
"Migliore amico" or (if we want to be funny) "Amicone" 😄

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 8]
```diff
@@ -0,0 +1,915 @@
...
+          <a href={`/edit/${friend.id}`}>Edit</a>
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:879]:
```suggestion
          <a href={`/edit/${friend.id}`}>Modifica</a>
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 9]
```diff
@@ -0,0 +1,915 @@
...
+
+### Creazione delle pagine
+
+Crea le pagine che utilizzeranno gli endpoint di Firebase:
+
+- `src/pages/register` - conterrà un form per registrare un utente
+- `src/pages/signin` - conterrà un form per effettuare l'accesso di un utente
+- `src/pages/dashboard` - conterrà una dashboard a cui solo gli utenti autenticati possono accedere
+
+L'esempio di `src/pages/register.astro` qui sotto include un form che invierà una richiesta `POST` al endpoint `/api/auth/register`. Questo endpoint creerà un nuovo utente utilizzando i dati del form e quindi reindirizzerà l'utente alla pagina `/signin`.
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:320]:
```suggestion
L'esempio di `src/pages/register.astro` qui sotto include un form che invierà una richiesta `POST` all'endpoint `/api/auth/register`. Questo endpoint creerà un nuovo utente utilizzando i dati del form e quindi reindirizzerà l'utente alla pagina `/signin`.
```

File: src/content/docs/it/guides/backend/google-firebase.mdx [hunk 10]
```diff
@@ -0,0 +1,915 @@
...
+    <label for="password">Password</label>
+    <input type="password" name="password" id="password" />
+    <button type="submit">Registrati</button>
+  </form>
+</Layout>
+```
+
+`src/pages/signin.astro` utilizza l'app server di Firebase per verificare il cookie per la sessione dell'utente. Se l'utente è autenticato, la pagina reindirizzerà l'utente alla pagina `/dashboard`.
+
+La pagina di esempio qui sotto contiene un form che invierà una richiesta `POST` al endpoint `/api/auth/signin` con il token ID generato dall'app client di Firebase.
```
Comment (csarnataro) [inline src/content/docs/it/guides/backend/google-firebase.mdx:344]:
```suggestion
La pagina di esempio qui sotto contiene un form che invierà una richiesta `POST` all'endpoint `/api/auth/signin` con il token ID generato dall'app client di Firebase.
```

**Review by csarnataro — APPROVED** [review]
LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks for the great work @vclemenzi and everyone reviewing 🙌

**Pull request merged**